common/app/conf/Filters.scala

```diff
@@ -67,6 +67,27 @@ object SurrogateKeyFilter extends Filter with ExecutionContexts {
   }
 }
 
+object AmpFilter extends Filter with ExecutionContexts with implicits.Requests {
+  override def apply(nextFilter: (RequestHeader) => Future[Result])(request: RequestHeader): Future[Result] = {
+    if (request.isAmp) {
+      val domain = request.headers.get("Origin") match {
+        case None => {
+          "https://" + request.domain
+        }
+        case Some(requestOrigin) => {
+          requestOrigin
+        }
+      }
```

Comment:
could you use a more idiomatic `request.headers.get("Origin").fold("https://" + request.domain)(_)` here?
Or `request.headers.get("Origin").map(_).getOrElse("https://"+request.domain)'

A good read about Option matching: http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/